joint_state_publisher/test/test_parsing.py

```diff
@@ -0,0 +1,736 @@
+
+    <joint name="j12" type="revolute">
+      <parent link="link1"/>
+      <child link="link2"/>
+      <limit effort="10" velocity="10" lower="-1"/>
+    </joint>
+  </robot>
+</urdf>"""
+
+    urdf_filename = tmpdir / 'revolute.urdf'
+    urdf_filename.write_text(urdf, encoding='utf-8')
+
+    jsp = joint_state_publisher.joint_state_publisher.JointStatePublisher(urdf_filename)
+
+    assert not jsp.free_joints
+
+    assert not jsp.joint_list
+
+    assert not jsp.dependent_joints
+
+
+def test_urdf_revolute_with_bogus_lower(tmpdir, rclpy_fixture):
+    urdf = """<?xml version="1.0"?>
+<urdf>
+  <robot name="multi_joint_robot">
+    <link name="link1"/>
+    <link name="link2"/>
+
+    <joint name="j12" type="revolute">
+      <parent link="link1"/>
+      <child link="link2"/>
+      <limit effort="10" velocity="10" lower="foo" upper="1"/>
+    </joint>
+  </robot>
+</urdf>"""
+
+    urdf_filename = tmpdir / 'revolute.urdf'
+    urdf_filename.write_text(urdf, encoding='utf-8')
+
+    jsp = joint_state_publisher.joint_state_publisher.JointStatePublisher(urdf_filename)
+
+    assert not jsp.free_joints
+
+    assert not jsp.joint_list
+
+    assert not jsp.dependent_joints
+
+
+def test_urdf_revolute_with_bogus_upper(tmpdir, rclpy_fixture):
+    urdf = """<?xml version="1.0"?>
+<urdf>
+  <robot name="multi_joint_robot">
+    <link name="link1"/>
+    <link name="link2"/>
+
+    <joint name="j12" type="revolute">
+      <parent link="link1"/>
+      <child link="link2"/>
+      <limit effort="10" velocity="10" lower="-1" upper="foo"/>
+    </joint>
+  </robot>
+</urdf>"""
+
+    urdf_filename = tmpdir / 'revolute.urdf'
+    urdf_filename.write_text(urdf, encoding='utf-8')
+
+    jsp = joint_state_publisher.joint_state_publisher.JointStatePublisher(urdf_filename)
+
+    assert not jsp.free_joints
+
+    assert not jsp.joint_list
+
+    assert not jsp.dependent_joints
+
+
+def test_collada_revolute(tmpdir, rclpy_fixture):
+    collada = """<?xml version="1.0" encoding="UTF-8"?>
+<COLLADA xmlns="http://www.collada.org/2008/03/COLLADASchema" version="1.5.0">
+  <library_kinematics_models id="kmodels">
+    <kinematics_model id="kmodel0" name="multi_joint_robot">
+      <technique_common>
+        <joint name="j12" sid="j12">
+          <revolute sid="axis0">
+            <axis>1 0 0</axis>
+            <limits>
+              <min>-57.29577951308232</min>
+              <max>57.29577951308232</max>
+            </limits>
+          </revolute>
+        </joint>
+      </technique_common>
+    </kinematics_model>
+  </library_kinematics_models>
+</COLLADA>"""
+
+    collada_filename = tmpdir / 'revolute.dae'
+    collada_filename.write_text(collada, encoding='utf-8')
+
+    jsp = joint_state_publisher.joint_state_publisher.JointStatePublisher(collada_filename)
+
+    assert len(jsp.free_joints) == 1
+    assert jsp.free_joints['j12']['max'] == pytest.approx(1.0)
+    assert jsp.free_joints['j12']['min'] == pytest.approx(-1.0)
```

Comment:
Limits don't match with the model

Reply:
Ahh scratch that, the parser is converting to radians

Reply:
Yes, exactly!